lib/modules/manager/sbt/extract.ts

```diff
@@ -323,23 +373,123 @@ export function extractPackageFile(
 
   try {
     parsedResult = scala.query(content, query, {
-      vars: {},
+      globalVars,
+      localVars,
       deps: [],
-      registryUrls: [REGISTRY_URLS.mavenCentral],
+      registryUrls,
+      packageFile,
+      scalaVersion,
     });
   } catch (err) /* istanbul ignore next */ {
-    logger.debug({ err, packageFile }, 'Sbt parsing error');
+    logger.warn({ err, packageFile }, 'Sbt parsing error');
   }
 
   if (!parsedResult) {
     return null;
   }
 
-  const { deps, packageFileVersion } = parsedResult;
+  delete parsedResult?.globalVars;
+  return parsedResult;
+}
 
-  if (!deps.length) {
-    return null;
+function prepareLoadPackageFiles(
+  packageFilesContent: { packageFile: string; content: string }[]
+): Pick<Ctx, 'globalVars' | 'registryUrls' | 'scalaVersion'> {
+  // Return variable
+  let globalVars: Ctx['globalVars'] = {};
+  let registryUrls: Ctx['registryUrls'] = [REGISTRY_URLS.mavenCentral];
+  let scalaVersion: Ctx['scalaVersion'] = undefined;
+
+  for (const { packageFile, content } of packageFilesContent) {
+    const res = extractDependency(content, {
```

Comment:
This is very inefficient because with `scala.query` in `extractDependency`, you also extract dependencies which you then throw away here. I'd rather make two separate queries, so that you don't match absolutely **everything** twice.

Reply:
That is exactly what I wanted. The first iteration is to store variables without this it wouldn't be able to resolve `Versions.akka` in this file https://github.com/renovatebot/renovate/pull/21622/files#diff-09d027666d4c679177a665d5920a343d2b1ef0c1cad4a4822a649fb991076910R35

You can take a look on `simple-project` without do it twice it would have a race condition on it

Reply:
I still don't have much idea to resolve the variable. Without populating it in the first loop. 
It's like we can do it all without 2 loops but we'll be missing some variables.

Reply:
Probably that was a misunderstanding but I didn't mean there should be only one loop. The idea was rather to have to distinct `scala.query` operations that each encompass different queries. In order words, split querying global vars from the rest.